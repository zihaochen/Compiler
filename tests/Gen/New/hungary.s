.data
	_printf_cnt: .space 4
	_e: .space 4
	_struct_member_1: .space 4
	_struct_member_2: .space 4
	_struct_member_3: .space 4
	_n: .space 4
	_m: .space 4
	_a: .space 4
	_b: .space 4
	_nxt: .space 4
	_visit: .space 4
	ASU_1: .space 20
	ASU_2: .space 44
	ASU_3: .space 44
	ASU_4: .space 44
	ASU_5: .space 44
	ASU_6: .space 44
	str_1: .space 16
.text
main:
	move $fp, $sp     # start using memory here
	move $gp, $sp     # set global pointer (unused)
	subu $sp, $sp, 4
	la $t0, ASU_1
	sw $t0, _e
	la $t0, ASU_2
	sw $t0, _struct_member_1
	la $t0, ASU_3
	sw $t0, _struct_member_2
	la $t0, ASU_4
	sw $t0, _struct_member_3
	la $t0, ASU_5
	sw $t0, _nxt
	la $t0, ASU_6
	sw $t0, _visit
	li $t0, 0
	sw $t0, _printf_cnt
	lw $t1, _e
	lw $t2, _struct_member_1
	sw $t2, 0($t1)
	lw $t1, _e
	lw $t2, _struct_member_2
	sw $t2, 4($t1)
	lw $t1, _e
	lw $t2, _struct_member_3
	sw $t2, 8($t1)
	li $t0, 37
	la $t1, str_1
	sw $t0, 0($t1)
	li $t0, 100
	la $t1, str_1
	sw $t0, 4($t1)
	li $t0, 10
	la $t1, str_1
	sw $t0, 8($t1)
	li $t0, 0
	la $t1, str_1
	sw $t0, 12($t1)
	j _main
_addEdge:
	sw $ra, ($sp)
	lw $t1, _e
	lw $t0, 12($t1)
	sw $t0, -8($sp)
	lw $t1, _e
	lw $t0, 0($t1)
	sw $t0, -12($sp)
	li $t0, 0
	sw $t0, -16($sp)
	li $t2, 4
	lw $t1, -8($sp)
	mul $t0, $t1, $t2
	sw $t0, -20($sp)
	lw $t2, -20($sp)
	lw $t1, -16($sp)
	add $t0, $t1, $t2
	sw $t0, -16($sp)
	lw $t2, -16($sp)
	lw $t1, -12($sp)
	add $t0, $t1, $t2
	sw $t0, -24($sp)
	lw $t0, -24($sp)
	lw $t1, 4($sp)
	sw $t1, ($t0)
	lw $t0, -24($sp)
	lw $t1, ($t0)
	sw $t1, -4($sp)
	lw $t0, -24($sp)
	lw $t1, ($t0)
	sw $t1, -4($sp)
	lw $t1, _e
	lw $t0, 8($t1)
	sw $t0, -36($sp)
	li $t0, 0
	sw $t0, -40($sp)
	li $t2, 4
	lw $t1, 8($sp)
	mul $t0, $t1, $t2
	sw $t0, -44($sp)
	lw $t2, -44($sp)
	lw $t1, -40($sp)
	add $t0, $t1, $t2
	sw $t0, -40($sp)
	lw $t2, -40($sp)
	lw $t1, -36($sp)
	add $t0, $t1, $t2
	sw $t0, -48($sp)
	lw $t0, -48($sp)
	lw $t1, ($t0)
	sw $t1, -52($sp)
	lw $t1, _e
	lw $t0, 12($t1)
	sw $t0, -56($sp)
	lw $t1, _e
	lw $t0, 4($t1)
	sw $t0, -60($sp)
	li $t0, 0
	sw $t0, -64($sp)
	li $t2, 4
	lw $t1, -56($sp)
	mul $t0, $t1, $t2
	sw $t0, -68($sp)
	lw $t2, -68($sp)
	lw $t1, -64($sp)
	add $t0, $t1, $t2
	sw $t0, -64($sp)
	lw $t2, -64($sp)
	lw $t1, -60($sp)
	add $t0, $t1, $t2
	sw $t0, -72($sp)
	lw $t0, -72($sp)
	lw $t1, -52($sp)
	sw $t1, ($t0)
	lw $t0, -72($sp)
	lw $t1, ($t0)
	sw $t1, -32($sp)
	lw $t0, -72($sp)
	lw $t1, ($t0)
	sw $t1, -32($sp)
	lw $t1, _e
	lw $t0, 12($t1)
	sw $t0, -92($sp)
	li $t2, 1
	lw $t1, -92($sp)
	add $t0, $t1, $t2
	sw $t0, -96($sp)
	lw $t1, _e
	lw $t2, -96($sp)
	sw $t2, 12($t1)
	li $t2, 1
	lw $t1, -88($sp)
	sub $t0, $t1, $t2
	sw $t0, -84($sp)
	lw $t1, _e
	lw $t0, 8($t1)
	sw $t0, -100($sp)
	li $t0, 0
	sw $t0, -104($sp)
	li $t2, 4
	lw $t1, 8($sp)
	mul $t0, $t1, $t2
	sw $t0, -108($sp)
	lw $t2, -108($sp)
	lw $t1, -104($sp)
	add $t0, $t1, $t2
	sw $t0, -104($sp)
	lw $t2, -104($sp)
	lw $t1, -100($sp)
	add $t0, $t1, $t2
	sw $t0, -112($sp)
	lw $t0, -112($sp)
	lw $t1, -84($sp)
	sw $t1, ($t0)
	lw $t0, -112($sp)
	lw $t1, ($t0)
	sw $t1, -80($sp)
	lw $t0, -112($sp)
	lw $t1, ($t0)
	sw $t1, -80($sp)
	jr $ra
_dfs:
	sw $ra, ($sp)
	lw $t1, _e
	lw $t0, 8($t1)
	sw $t0, -16($sp)
	li $t0, 0
	sw $t0, -20($sp)
	li $t2, 4
	lw $t1, 4($sp)
	mul $t0, $t1, $t2
	sw $t0, -24($sp)
	lw $t2, -24($sp)
	lw $t1, -20($sp)
	add $t0, $t1, $t2
	sw $t0, -20($sp)
	lw $t2, -20($sp)
	lw $t1, -16($sp)
	add $t0, $t1, $t2
	sw $t0, -28($sp)
	lw $t0, -28($sp)
	lw $t1, ($t0)
	sw $t1, -32($sp)
	lw $t0, -32($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -12($sp)
L2:
	lw $t2, -4($sp)
	not $t0, $t2
	sw $t0, -36($sp)
	lw $t0, -36($sp)
	bne $t0, $0, L3
	b L1
L3:
	lw $t1, _e
	lw $t0, 0($t1)
	sw $t0, -44($sp)
	li $t0, 0
	sw $t0, -48($sp)
	li $t2, 4
	lw $t1, -4($sp)
	mul $t0, $t1, $t2
	sw $t0, -52($sp)
	lw $t2, -52($sp)
	lw $t1, -48($sp)
	add $t0, $t1, $t2
	sw $t0, -48($sp)
	lw $t2, -48($sp)
	lw $t1, -44($sp)
	add $t0, $t1, $t2
	sw $t0, -56($sp)
	lw $t0, -56($sp)
	lw $t1, ($t0)
	sw $t1, -60($sp)
	lw $t0, -60($sp)
	sw $t0, -8($sp)
	lw $t0, -8($sp)
	sw $t0, -40($sp)
	li $t0, 0
	sw $t0, -64($sp)
	li $t2, 4
	lw $t1, -8($sp)
	mul $t0, $t1, $t2
	sw $t0, -68($sp)
	lw $t2, -68($sp)
	lw $t1, -64($sp)
	add $t0, $t1, $t2
	sw $t0, -64($sp)
	lw $t2, -64($sp)
	lw $t1, _visit
	add $t0, $t1, $t2
	sw $t0, -72($sp)
	lw $t0, -72($sp)
	lw $t1, ($t0)
	sw $t1, -76($sp)
	lw $t0, -76($sp)
	bne $t0, $0, L5
	b L4
L5:
	b L2
L4:
	li $t0, 0
	sw $t0, -84($sp)
	li $t2, 4
	lw $t1, -8($sp)
	mul $t0, $t1, $t2
	sw $t0, -88($sp)
	lw $t2, -88($sp)
	lw $t1, -84($sp)
	add $t0, $t1, $t2
	sw $t0, -84($sp)
	lw $t2, -84($sp)
	lw $t1, _visit
	add $t0, $t1, $t2
	sw $t0, -92($sp)
	lw $t0, -92($sp)
	li $t1, 1
	sw $t1, ($t0)
	lw $t0, -92($sp)
	lw $t1, ($t0)
	sw $t1, -80($sp)
	lw $t0, -92($sp)
	lw $t1, ($t0)
	sw $t1, -80($sp)
	li $t0, 0
	sw $t0, -100($sp)
	li $t2, 4
	lw $t1, -8($sp)
	mul $t0, $t1, $t2
	sw $t0, -104($sp)
	lw $t2, -104($sp)
	lw $t1, -100($sp)
	add $t0, $t1, $t2
	sw $t0, -100($sp)
	lw $t2, -100($sp)
	lw $t1, _nxt
	add $t0, $t1, $t2
	sw $t0, -108($sp)
	lw $t0, -108($sp)
	lw $t1, ($t0)
	sw $t1, -112($sp)
	li $t2, -1
	lw $t1, -112($sp)
	seq $t0, $t1, $t2
	sw $t0, -116($sp)
	lw $t0, -116($sp)
	bne $t0, $0, L7
	b L8
L8:
	lw $t0, -8($sp)
	sw $t0, -168($sp)
	subu $sp, $sp, 172
	jal _dfs
	addi $sp, $sp, 172
	sw $v0, -120($sp)
	lw $ra, ($sp)
	lw $t0, -120($sp)
	bne $t0, $0, L7
	b L6
L7:
	li $t0, 0
	sw $t0, -128($sp)
	li $t2, 4
	lw $t1, -8($sp)
	mul $t0, $t1, $t2
	sw $t0, -132($sp)
	lw $t2, -132($sp)
	lw $t1, -128($sp)
	add $t0, $t1, $t2
	sw $t0, -128($sp)
	lw $t2, -128($sp)
	lw $t1, _nxt
	add $t0, $t1, $t2
	sw $t0, -136($sp)
	lw $t0, -136($sp)
	lw $t1, 4($sp)
	sw $t1, ($t0)
	lw $t0, -136($sp)
	lw $t1, ($t0)
	sw $t1, -124($sp)
	lw $t0, -136($sp)
	lw $t1, ($t0)
	sw $t1, -124($sp)
	li $v0, 1
	sw $v0, 0($sp)
	jr $ra
L6:
	lw $t1, _e
	lw $t0, 4($t1)
	sw $t0, -148($sp)
	li $t0, 0
	sw $t0, -152($sp)
	li $t2, 4
	lw $t1, -4($sp)
	mul $t0, $t1, $t2
	sw $t0, -156($sp)
	lw $t2, -156($sp)
	lw $t1, -152($sp)
	add $t0, $t1, $t2
	sw $t0, -152($sp)
	lw $t2, -152($sp)
	lw $t1, -148($sp)
	add $t0, $t1, $t2
	sw $t0, -160($sp)
	lw $t0, -160($sp)
	lw $t1, ($t0)
	sw $t1, -164($sp)
	lw $t0, -164($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -144($sp)
	b L2
L1:
	li $v0, 0
	sw $v0, 0($sp)
	jr $ra
	jr $ra
_getInt:
	sw $ra, ($sp)
	li $t0, 0
	sw $t0, -8($sp)
	subu $sp, $sp, 68
	jal _getchar
	addi $sp, $sp, 68
	sw $v0, -16($sp)
	lw $ra, ($sp)
	lw $t0, -16($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -12($sp)
L10:
	li $t2, 48
	lw $t1, -4($sp)
	slt $t0, $t1, $t2
	sw $t0, -20($sp)
	lw $t0, -20($sp)
	bne $t0, $0, L11
	b L12
L12:
	li $t2, 57
	lw $t1, -4($sp)
	sgt $t0, $t1, $t2
	sw $t0, -24($sp)
	lw $t0, -24($sp)
	bne $t0, $0, L11
	b L9
L11:
	subu $sp, $sp, 68
	jal _getchar
	addi $sp, $sp, 68
	sw $v0, -32($sp)
	lw $ra, ($sp)
	lw $t0, -32($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -28($sp)
	b L10
L9:
L14:
	li $t2, 48
	lw $t1, -4($sp)
	sge $t0, $t1, $t2
	sw $t0, -36($sp)
	lw $t0, -36($sp)
	bne $t0, $0, L16
	b L13
L16:
	li $t2, 57
	lw $t1, -4($sp)
	sle $t0, $t1, $t2
	sw $t0, -40($sp)
	lw $t0, -40($sp)
	bne $t0, $0, L15
	b L13
L15:
	li $t2, 10
	lw $t1, -8($sp)
	mul $t0, $t1, $t2
	sw $t0, -48($sp)
	lw $t2, -4($sp)
	lw $t1, -48($sp)
	add $t0, $t1, $t2
	sw $t0, -52($sp)
	li $t2, 48
	lw $t1, -52($sp)
	sub $t0, $t1, $t2
	sw $t0, -56($sp)
	lw $t0, -56($sp)
	sw $t0, -8($sp)
	lw $t0, -8($sp)
	sw $t0, -44($sp)
	subu $sp, $sp, 68
	jal _getchar
	addi $sp, $sp, 68
	sw $v0, -64($sp)
	lw $ra, ($sp)
	lw $t0, -64($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -60($sp)
	b L14
L13:
	lw $v0, -8($sp)
	sw $v0, -8($sp)
	jr $ra
	jr $ra
_main:
	sw $ra, ($sp)
	la $t0, str_1
	sw $t0, -228($sp)
	li $t0, 0
	sw $t0, -12($sp)
	subu $sp, $sp, 236
	jal _getInt
	addi $sp, $sp, 236
	sw $v0, -20($sp)
	lw $ra, ($sp)
	lw $t0, -20($sp)
	sw $t0, _n
	lw $t0, _n
	sw $t0, -16($sp)
	subu $sp, $sp, 236
	jal _getInt
	addi $sp, $sp, 236
	sw $v0, -28($sp)
	lw $ra, ($sp)
	lw $t0, -28($sp)
	sw $t0, _m
	lw $t0, _m
	sw $t0, -24($sp)
	lw $t1, _e
	li $t2, 0
	sw $t2, 12($t1)
	li $t0, 0
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -36($sp)
L18:
	li $t2, 10
	lw $t1, -4($sp)
	slt $t0, $t1, $t2
	sw $t0, -40($sp)
	lw $t0, -40($sp)
	bne $t0, $0, L19
	b L17
L19:
	lw $t1, _e
	lw $t0, 8($t1)
	sw $t0, -48($sp)
	li $t0, 0
	sw $t0, -52($sp)
	li $t2, 4
	lw $t1, -4($sp)
	mul $t0, $t1, $t2
	sw $t0, -56($sp)
	lw $t2, -56($sp)
	lw $t1, -52($sp)
	add $t0, $t1, $t2
	sw $t0, -52($sp)
	lw $t2, -52($sp)
	lw $t1, -48($sp)
	add $t0, $t1, $t2
	sw $t0, -60($sp)
	lw $t0, -60($sp)
	li $t1, -1
	sw $t1, ($t0)
	lw $t0, -60($sp)
	lw $t1, ($t0)
	sw $t1, -44($sp)
	lw $t0, -60($sp)
	lw $t1, ($t0)
	sw $t1, -44($sp)
	li $t2, 1
	lw $t1, -4($sp)
	add $t0, $t1, $t2
	sw $t0, -76($sp)
	lw $t0, -76($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -72($sp)
	li $t2, 1
	lw $t1, -72($sp)
	sub $t0, $t1, $t2
	sw $t0, -68($sp)
	b L18
L17:
	li $t0, 1
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -80($sp)
L21:
	lw $t2, _m
	lw $t1, -4($sp)
	sle $t0, $t1, $t2
	sw $t0, -84($sp)
	lw $t0, -84($sp)
	bne $t0, $0, L22
	b L20
L22:
	subu $sp, $sp, 236
	jal _getInt
	addi $sp, $sp, 236
	sw $v0, -92($sp)
	lw $ra, ($sp)
	lw $t0, -92($sp)
	sw $t0, _a
	lw $t0, _a
	sw $t0, -88($sp)
	subu $sp, $sp, 236
	jal _getInt
	addi $sp, $sp, 236
	sw $v0, -100($sp)
	lw $ra, ($sp)
	lw $t0, -100($sp)
	sw $t0, _b
	lw $t0, _b
	sw $t0, -96($sp)
	lw $t0, _a
	sw $t0, -236($sp)
	lw $t0, _b
	sw $t0, -240($sp)
	subu $sp, $sp, 244
	jal _addEdge
	addi $sp, $sp, 244
	sw $v0, -104($sp)
	lw $ra, ($sp)
	li $t2, 1
	lw $t1, -4($sp)
	add $t0, $t1, $t2
	sw $t0, -116($sp)
	lw $t0, -116($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -112($sp)
	li $t2, 1
	lw $t1, -112($sp)
	sub $t0, $t1, $t2
	sw $t0, -108($sp)
	b L21
L20:
	li $t0, 1
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -120($sp)
L24:
	lw $t2, _n
	lw $t1, -4($sp)
	sle $t0, $t1, $t2
	sw $t0, -124($sp)
	lw $t0, -124($sp)
	bne $t0, $0, L25
	b L23
L25:
	li $t0, 0
	sw $t0, -132($sp)
	li $t2, 4
	lw $t1, -4($sp)
	mul $t0, $t1, $t2
	sw $t0, -136($sp)
	lw $t2, -136($sp)
	lw $t1, -132($sp)
	add $t0, $t1, $t2
	sw $t0, -132($sp)
	lw $t2, -132($sp)
	lw $t1, _nxt
	add $t0, $t1, $t2
	sw $t0, -140($sp)
	lw $t0, -140($sp)
	li $t1, -1
	sw $t1, ($t0)
	lw $t0, -140($sp)
	lw $t1, ($t0)
	sw $t1, -128($sp)
	lw $t0, -140($sp)
	lw $t1, ($t0)
	sw $t1, -128($sp)
	li $t2, 1
	lw $t1, -4($sp)
	add $t0, $t1, $t2
	sw $t0, -152($sp)
	lw $t0, -152($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -148($sp)
	b L24
L23:
	li $t0, 1
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -156($sp)
L27:
	lw $t2, _n
	lw $t1, -4($sp)
	sle $t0, $t1, $t2
	sw $t0, -160($sp)
	lw $t0, -160($sp)
	bne $t0, $0, L28
	b L26
L28:
	li $t0, 1
	sw $t0, -8($sp)
	lw $t0, -8($sp)
	sw $t0, -164($sp)
L30:
	lw $t2, _n
	lw $t1, -8($sp)
	sle $t0, $t1, $t2
	sw $t0, -168($sp)
	lw $t0, -168($sp)
	bne $t0, $0, L31
	b L29
L31:
	li $t0, 0
	sw $t0, -176($sp)
	li $t2, 4
	lw $t1, -8($sp)
	mul $t0, $t1, $t2
	sw $t0, -180($sp)
	lw $t2, -180($sp)
	lw $t1, -176($sp)
	add $t0, $t1, $t2
	sw $t0, -176($sp)
	lw $t2, -176($sp)
	lw $t1, _visit
	add $t0, $t1, $t2
	sw $t0, -184($sp)
	lw $t0, -184($sp)
	li $t1, 0
	sw $t1, ($t0)
	lw $t0, -184($sp)
	lw $t1, ($t0)
	sw $t1, -172($sp)
	lw $t0, -184($sp)
	lw $t1, ($t0)
	sw $t1, -172($sp)
	li $t2, 1
	lw $t1, -8($sp)
	add $t0, $t1, $t2
	sw $t0, -200($sp)
	lw $t0, -200($sp)
	sw $t0, -8($sp)
	lw $t0, -8($sp)
	sw $t0, -196($sp)
	li $t2, 1
	lw $t1, -196($sp)
	sub $t0, $t1, $t2
	sw $t0, -192($sp)
	b L30
L29:
	lw $t0, -4($sp)
	sw $t0, -236($sp)
	subu $sp, $sp, 240
	jal _dfs
	addi $sp, $sp, 240
	sw $v0, -208($sp)
	lw $ra, ($sp)
	lw $t2, -208($sp)
	lw $t1, -12($sp)
	add $t0, $t1, $t2
	sw $t0, -212($sp)
	lw $t0, -212($sp)
	sw $t0, -12($sp)
	lw $t0, -12($sp)
	sw $t0, -204($sp)
	li $t2, 1
	lw $t1, -4($sp)
	add $t0, $t1, $t2
	sw $t0, -224($sp)
	lw $t0, -224($sp)
	sw $t0, -4($sp)
	lw $t0, -4($sp)
	sw $t0, -220($sp)
	li $t2, 1
	lw $t1, -220($sp)
	sub $t0, $t1, $t2
	sw $t0, -216($sp)
	b L27
L26:
	lw $t0, -228($sp)
	sw $t0, -236($sp)
	lw $t0, -12($sp)
	sw $t0, -240($sp)
	li $t0, 2
	sw $t0, _printf_cnt
	subu $sp, $sp, 244
	jal _printf
	addi $sp, $sp, 244
	sw $v0, -232($sp)
	lw $ra, ($sp)
	li $v0, 0
	sw $v0, 0($sp)
	jr $ra
	jr $ra

_printf:
	lw $t0, _printf_cnt
	mul $t0, $t0, 4
	add $t0, $t0, $sp
	lw $a1, ($t0)
	move $a2, $t0
	
_printf_loop:
	lb $a0, 0($a1)
	beq $a0, 0, _printf_end
	addu $a1, $a1, 4
	beq $a0, '%', _printf_fmt
	li $v0, 11
	syscall
	b _printf_loop

_printf_fmt:
	lb $a0, 0($a1)
	addu $a1, $a1, 4
	beq $a0, 'd', _printf_int
	beq $a0, 's', _printf_str
	beq $a0, 'c', _printf_char
	beq $a0, '0', _printf_width
	beq $a0, '.', _printf_width

_printf_int:
	subu $a2, $a2, 4
	lw $a0, 0($a2)
	li $v0, 1
	syscall
	b _printf_loop

_printf_str:
	subu $a2, $a2, 4
	lw $a3, 0($a2)
	b _printf_str_loop

_printf_str_loop:
	addu $a3, $a3, 4
	lw $a0, ($a3)
	beq $a0, 0, _printf_loop
	li $v0, 11
	syscall
	b _printf_str_loop

_printf_char:
	subu $a2, $a2, 4
	lw $a0, 0($a2)
	li $v0, 11
	syscall
	b _printf_loop

_printf_width:
	lb $t1, 0($a1)
	addu $a1, $a1, 4
	sub $t1, $t1, '1'
	addu $a1, $a1, 4
	subu $a2, $a2, 4
	lw $t0, 0($a2)
	blt $t0, 10, _printf_width_tmp
	subu $t1, $t1, 1
	blt $t0, 100, _printf_width_tmp
	subu $t1, $t1, 1
	blt $t0, 1000, _printf_width_tmp
	subu $t1, $t1, 1
	blt $t0, 10000, _printf_width_tmp
	subu $t1, $t1, 1
	blt $t0, 100000, _printf_width_tmp
	subu $t1, $t1, 1
	blt $t0, 1000000, _printf_width_tmp
	subu $t1, $t1, 1
	blt $t0, 10000000, _printf_width_tmp
	subu $t1, $t1, 1
	blt $t0, 100000000, _printf_width_tmp
	subu $t1, $t1, 1

_printf_width_tmp:
	li $a0, '0'
	beq $t1, 0, _printf_width_tmp2
	subu $t1, $t1, 1
	li $v0, 11
	syscall
	b _printf_width_tmp

_printf_width_tmp2:
	move $a0, $t0
	li $v0, 1
	syscall
	b _printf_loop

_printf_end:
	jr $ra

_getchar:
	li $v0, 12
	syscall
	jr $ra

_malloc_struct:
	lw $a0, 4($sp)
	li $v0, 9
	syscall
	move $t0, $v0
	li $v0, 9
	li $a0, 4
	syscall
	sw $t0, ($v0)
	jr $ra

_malloc:
	lw $a0, 4($sp)
	li $v0, 9
	syscall
	jr $ra
