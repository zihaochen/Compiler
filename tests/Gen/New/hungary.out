_start: 
_printf_cnt = 0
_e[0] = _struct_member_1 , 4
_e[4] = _struct_member_2 , 4
_e[8] = _struct_member_3 , 4

_malloc: 

_getchar: 

_printf: 

_addEdge: 
Temp2 = _e[12] , 4
Temp3 = _e[0] , 40
Temp4 = 0
Temp5 = Temp2 MUL 4
Temp4 = Temp4 ADD Temp5
Temp6 = Temp3 ADD Temp4
*Temp6 = b , 4
Temp1 = *Temp6, 4
Temp1 = *Temp6, 4
Temp9 = _e[8] , 40
Temp10 = 0
Temp11 = a MUL 4
Temp10 = Temp10 ADD Temp11
Temp12 = Temp9 ADD Temp10
Temp13 = *Temp12, 4
Temp14 = _e[12] , 4
Temp15 = _e[4] , 40
Temp16 = 0
Temp17 = Temp14 MUL 4
Temp16 = Temp16 ADD Temp17
Temp18 = Temp15 ADD Temp16
*Temp18 = Temp13 , 4
Temp8 = *Temp18, 4
Temp8 = *Temp18, 4
Temp23 = _e[12] , 4
Temp24 = Temp23 ADD 1
_e[12] = Temp24 , 4
Temp21 = Temp22 SUB 1
Temp25 = _e[8] , 40
Temp26 = 0
Temp27 = a MUL 4
Temp26 = Temp26 ADD Temp27
Temp28 = Temp25 ADD Temp26
*Temp28 = Temp21 , 4
Temp20 = *Temp28, 4
Temp20 = *Temp28, 4

_dfs: 
Temp31 = _e[8] , 40
Temp32 = 0
Temp33 = x MUL 4
Temp32 = Temp32 ADD Temp33
Temp34 = Temp31 ADD Temp32
Temp35 = *Temp34, 4
_i = Temp35
Temp30 = _i
Label :2	Temp36 =  TILDE _i
If Not Equal to Zero Temp36 Goto Label :3	
Goto label:1
Label :3	Temp38 = _e[0] , 40
Temp39 = 0
Temp40 = _i MUL 4
Temp39 = Temp39 ADD Temp40
Temp41 = Temp38 ADD Temp39
Temp42 = *Temp41, 4
_y = Temp42
Temp37 = _y
Temp43 = 0
Temp44 = _y MUL 4
Temp43 = Temp43 ADD Temp44
Temp45 = _visit ADD Temp43
Temp46 = *Temp45, 4
If Not Equal to Zero Temp46 Goto Label :5	
Goto label:4
Label :5	Goto label:2
Label :4	Temp48 = 0
Temp49 = _y MUL 4
Temp48 = Temp48 ADD Temp49
Temp50 = _visit ADD Temp48
*Temp50 = 1 , 4
Temp47 = *Temp50, 4
Temp47 = *Temp50, 4
Temp52 = 0
Temp53 = _y MUL 4
Temp52 = Temp52 ADD Temp53
Temp54 = _nxt ADD Temp52
Temp55 = *Temp54, 4
Temp56 = Temp55 EQ -1
If Not Equal to Zero Temp56 Goto Label :7	
Goto label:8
Label :8	BasicParam _y
Temp57 = call _dfs , 1
If Not Equal to Zero Temp57 Goto Label :7	
Goto label:6
Label :7	Temp59 = 0
Temp60 = _y MUL 4
Temp59 = Temp59 ADD Temp60
Temp61 = _nxt ADD Temp59
*Temp61 = x , 4
Temp58 = *Temp61, 4
Temp58 = *Temp61, 4
Return BasicParam 1
Label :6	Temp64 = _e[4] , 40
Temp65 = 0
Temp66 = _i MUL 4
Temp65 = Temp65 ADD Temp66
Temp67 = Temp64 ADD Temp65
Temp68 = *Temp67, 4
_i = Temp68
Temp63 = _i
Goto label:2
Label :1	Return BasicParam 0

_getInt: 
_x = 0
Temp70 = call _getchar , 0
_ch = Temp70
Temp69 = _ch
Label :10	Temp71 = _ch LT 48
If Not Equal to Zero Temp71 Goto Label :11	
Goto label:12
Label :12	Temp72 = _ch GT 57
If Not Equal to Zero Temp72 Goto Label :11	
Goto label:9
Label :11	Temp74 = call _getchar , 0
_ch = Temp74
Temp73 = _ch
Goto label:10
Label :9	Label :14	Temp75 = _ch GE 48
If Not Equal to Zero Temp75 Goto Label :16	
Goto label:13
Label :16	Temp76 = _ch LE 57
If Not Equal to Zero Temp76 Goto Label :15	
Goto label:13
Label :15	Temp78 = _x MUL 10
Temp79 = Temp78 ADD _ch
Temp80 = Temp79 SUB 48
_x = Temp80
Temp77 = _x
Temp82 = call _getchar , 0
_ch = Temp82
Temp81 = _ch
Goto label:14
Label :13	Return BasicParam _x

_main: 
_answer = 0
Temp84 = call _getInt , 0
_n = Temp84
Temp83 = _n
Temp86 = call _getInt , 0
_m = Temp86
Temp85 = _m
_e[12] = 0 , 4
_i = 0
Temp88 = _i
Label :18	Temp89 = _i LT 10
If Not Equal to Zero Temp89 Goto Label :19	
Goto label:17
Label :19	Temp91 = _e[8] , 40
Temp92 = 0
Temp93 = _i MUL 4
Temp92 = Temp92 ADD Temp93
Temp94 = Temp91 ADD Temp92
*Temp94 = -1 , 4
Temp90 = *Temp94, 4
Temp90 = *Temp94, 4
Temp98 = _i ADD 1
_i = Temp98
Temp97 = _i
Temp96 = Temp97 SUB 1
Goto label:18
Label :17	_i = 1
Temp99 = _i
Label :21	Temp100 = _i LE _m
If Not Equal to Zero Temp100 Goto Label :22	
Goto label:20
Label :22	Temp102 = call _getInt , 0
_a = Temp102
Temp101 = _a
Temp104 = call _getInt , 0
_b = Temp104
Temp103 = _b
BasicParam _a
BasicParam _b
Temp105 = call _addEdge , 2
Temp108 = _i ADD 1
_i = Temp108
Temp107 = _i
Temp106 = Temp107 SUB 1
Goto label:21
Label :20	_i = 1
Temp109 = _i
Label :24	Temp110 = _i LE _n
If Not Equal to Zero Temp110 Goto Label :25	
Goto label:23
Label :25	Temp112 = 0
Temp113 = _i MUL 4
Temp112 = Temp112 ADD Temp113
Temp114 = _nxt ADD Temp112
*Temp114 = -1 , 4
Temp111 = *Temp114, 4
Temp111 = *Temp114, 4
Temp117 = _i ADD 1
_i = Temp117
Temp116 = _i
Goto label:24
Label :23	_i = 1
Temp118 = _i
Label :27	Temp119 = _i LE _n
If Not Equal to Zero Temp119 Goto Label :28	
Goto label:26
Label :28	_j = 1
Temp120 = _j
Label :30	Temp121 = _j LE _n
If Not Equal to Zero Temp121 Goto Label :31	
Goto label:29
Label :31	Temp123 = 0
Temp124 = _j MUL 4
Temp123 = Temp123 ADD Temp124
Temp125 = _visit ADD Temp123
*Temp125 = 0 , 4
Temp122 = *Temp125, 4
Temp122 = *Temp125, 4
Temp129 = _j ADD 1
_j = Temp129
Temp128 = _j
Temp127 = Temp128 SUB 1
Goto label:30
Label :29	BasicParam _i
Temp131 = call _dfs , 1
Temp132 = _answer ADD Temp131
_answer = Temp132
Temp130 = _answer
Temp135 = _i ADD 1
_i = Temp135
Temp134 = _i
Temp133 = Temp134 SUB 1
Goto label:27
Label :26	BasicParam Temp136
BasicParam _answer
_printf_cnt = 2
Temp137 = call _printf , 2
Return BasicParam 0

